src/runtime/spec-interfaces.ts

```diff
@@ -0,0 +1,62 @@
+/**
+ * @license
+ * Copyright (c) 2019 Google Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {DirectionPreSlandles} from './manifest-ast-nodes.js';
+import {Claim} from './particle-claim.js';
+import {Type} from './type.js';
+
+export type CheckTarget = HandleConnectionSpecIntf | ProvideSlotConnectionSpecIntf;
+
+export interface CheckIntf {
+  target: CheckTarget;
+  // tslint:disable-next-line: no-any
+  expression: any;
```

Comment:
Is this not `CheckExpression` because it makes a cycle? I think we could use an interface for the expressions here.

Reply:
Maybe, but my job here is to kill the cycles we already have, not fix up the type-safety issues introduced when someone else tried to avoid cycles. Making this type safe is a worthy goal, but not one I want to take on in this PR. I will add a TODO.

Reply:
My quibble is that this is introducing a new instance of potentially unsafe data access, but I have given LGTM and meant it.

Reply:
Oh, sorry. I forgot that I introduced that, and not because it makes a cycle, but because it's really hard to define an interface for CheckExpression. It's a union type of a lot of things that all have different public members. But I'll take a look at it tomorrow and see what I can do. You're right that this is just trading one problem for another.

Reply:
Ok, I found a way that cleans this up and ends up simpler. Thanks for pointing out the flaw.